src/pages/workspaces/workspace/ShareWorkspaceModal/Collaborator.ts

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash/fp';
+import React, { Dispatch, SetStateAction } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { LabeledCheckbox, Link, Select, SelectProps } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { getPopupRoot } from 'src/components/popup-utils';
+import colors from 'src/libs/colors';
+import { getUser } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { BaseWorkspace, isAzureWorkspace, WorkspaceAccessLevel } from 'src/libs/workspace-utils';
+import { AccessEntry, WorkspaceAcl } from 'src/pages/workspaces/workspace/WorkspaceAcl';
+
+interface CollaboratorProps {
+  aclItem: AccessEntry; // the item to render
+  acl: WorkspaceAcl; // the entire current acl
+  setAcl: Dispatch<SetStateAction<WorkspaceAcl>>;
+  originalAcl: WorkspaceAcl; // the original acl, to determine new items
+  workspace: BaseWorkspace; // the workspace the acl belongs to
+  lastAddedEmail?: string; // the most recently added email to the list
+}
+
+/**
+ *  Note: the select menu of each Collaborator item NOT focusable via keyboard
+ *  tab will only focus on the remove buttons
+ */
+export const Collaborator = ({ originalAcl, aclItem, acl, setAcl, workspace, lastAddedEmail }: CollaboratorProps) => {
+  const { email, accessLevel, pending } = aclItem;
+  const POAccessLevel = 'PROJECT_OWNER';
```

Comment:
Shouldn't the type of this be declared (WorkspaceAccessLevel)?

Reply:
The transpiler will infer that it's compatible with the type, and allow it to be treated as such anywhere it's used.

It does add a bit of clarity though.